File: src/main/java/me/jsinco/jobsaddons/JobsAddons.java
package me.jsinco.jobsaddons;

import me.jsinco.jobsaddons.events.Listeners;
import me.jsinco.jobsaddons.perks.MiscPerkCommands;
import me.jsinco.jobsaddons.perks.JobsPerksCommand;
import me.jsinco.jobsaddons.perks.PotionCommands;
import org.bukkit.plugin.java.JavaPlugin;

public final class JobsAddons extends JavaPlugin {

    @Override
    public void onEnable() {
        if (!getDataFolder().exists()) getDataFolder().mkdir();
        FileManager fileManager = new FileManager(this);
        fileManager.setupConfig();
        fileManager.setupJobPerksFile();

        getServer().getPluginManager().registerEvents(new Listeners(this), this);

        new MiscPerkCommands(this); // Register commands
        new PotionCommands(this); // Register commands
        getCommand("jobperks").setExecutor(new JobsPerksCommand());
    }

    @Override
    public void onDisable() {
        // Plugin shutdown logic
    }
}

File: src/main/java/me/jsinco/jobsaddons/JobsAddons.kt
package me.jsinco.jobsaddons

import me.jsinco.jobsaddons.events.Listeners
import me.jsinco.jobsaddons.commands.JobsPerksCommand
import me.jsinco.jobsaddons.perks.MiscPerkCommands
import me.jsinco.jobsaddons.perks.PotionCommands
import me.jsinco.jobsaddons.util.ColorUtils
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.plugin.java.JavaPlugin

class JobsAddons : JavaPlugin() {
    override fun onEnable() {
        if (!dataFolder.exists()) dataFolder.mkdir()
        val fileManager = FileManager(this)
        fileManager.setupConfig()
        fileManager.setupJobPerksFile()
        server.pluginManager.registerEvents(Listeners(this), this)
        MiscPerkCommands(this) // Register commands
        PotionCommands(this) // Register commands
        getCommand("jobperks")!!.setExecutor(JobsPerksCommand(this))
        getCommand("jobsaddonsreload")!!.setExecutor(this)
    }

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        reloadConfig()
        sender.sendMessage(ColorUtils.colorcode("${config.getString("prefix")} Reloaded config"))
        return true
    }
}
